backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/carousel/internal/BpkCarouselImpl.kt

```diff
@@ -0,0 +1,90 @@
+/**
+ * Backpack for Android - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.skyscanner.backpack.compose.carousel.internal
+
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.BoxScope
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.snapshotFlow
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.testTag
+import com.google.accompanist.pager.ExperimentalPagerApi
+import com.google.accompanist.pager.HorizontalPager
+import com.google.accompanist.pager.rememberPagerState
+import kotlinx.coroutines.flow.distinctUntilChanged
+import net.skyscanner.backpack.compose.pageindicator.BpkPageIndicator
+import net.skyscanner.backpack.compose.pageindicator.BpkPageIndicatorStyle
+
+@OptIn(ExperimentalPagerApi::class)
+@Composable
+internal fun BpkCarouselImpl(
+  imageContent: @Composable (BoxScope.(Int) -> Unit),
+  count: Int,
+  currentImage: Int,
+  modifier: Modifier = Modifier,
+  onImageChanged: ((Int) -> Unit)? = null,
+) {
+  Box(modifier = modifier) {
+    val initialPageIndex = (Int.MAX_VALUE / 2) + currentImage
+    val (selectedPage, setSelectedPage) = rememberSaveable { mutableStateOf(currentImage) }
```

Comment:
Why do we remember this internally? It seems these props need to be injected and be a part of the component API

Reply:
Also, we just need to use by keyword with remembers here

Reply:
I remove this variable completely and use the currentPage from the pagerstate